cmd/artifact_install.go

```diff
@@ -0,0 +1,192 @@
+)
+
+const (
+	defaultPluginsDir    = "/usr/share/falco/plugins"
+	defaultRulesfilesDir = "/etc/falco"
+)
+
+type artifactInstallOptions struct {
+	*options.CommonOptions
+	credentialStore *authn.Store
+	rulesfilesDir   string
+	pluginsDir      string
+}
+
+// NewArtifactInstallCmd returns the artifact search command.
+func NewArtifactInstallCmd(ctx context.Context, opt *options.CommonOptions) *cobra.Command {
+	o := artifactInstallOptions{
+		CommonOptions: opt,
+	}
+
+	cmd := &cobra.Command{
+		Use:                   "install [ref1 [ref2 ...]] [flags]",
+		DisableFlagsInUseLine: true,
+		Short:                 "Install a list of artifacts",
+		Long:                  "Install a list of artifacts",
+		Args:                  cobra.MinimumNArgs(1),
+		Run: func(cmd *cobra.Command, args []string) {
+			o.Printer.CheckErr(o.RunArtifactInstall(ctx, args))
+		},
+	}
+
+	cmd.Flags().StringVarP(&o.rulesfilesDir, "rulesfiles-dir", "", defaultRulesfilesDir,
+		"directory where to install rules. Defaults to /etc/falco")
+	cmd.Flags().StringVarP(&o.pluginsDir, "plugins-dir", "", defaultPluginsDir,
+		"directory where to install plugins. Defaults to /usr/share/falco/plugins")
+
+	return cmd
+}
+
+// RunArtifactInstall executes the business logic for the artifact install command.
+func (o *artifactInstallOptions) RunArtifactInstall(ctx context.Context, args []string) error {
+	o.Printer.Info.Printfln("Reading all configured index files from %q", indexesFile)
+	indexConfig, err := index.NewConfig(indexesFile)
+	if err != nil {
+		return err
+	}
+
+	var allIndexes []*index.Index
+
+	o.Printer.Info.Println("Loading index files ...")
+	for _, indexConfigEntry := range indexConfig.Configs {
+		nameYaml := fmt.Sprintf("%s%s", indexConfigEntry.Name, ".yaml")
+		o.Printer.Verbosef("Loading index: %q", nameYaml)
+		i := index.New(indexConfigEntry.Name)
+		err := i.Read(filepath.Join(falcoctlPath, nameYaml))
+		if err != nil {
+			return fmt.Errorf("cannot load index %s: %w", i.Name, err)
+		}
+		allIndexes = append(allIndexes, i)
+	}
+
+	o.Printer.Info.Println("Merging all configured indexes ...")
+	mergedIndexes := index.NewMergedIndexes()
+	mergedIndexes.Merge(allIndexes...)
+	o.Printer.Verbosef("All configured indexes have been merged: %d", len(allIndexes))
+
+	o.credentialStore, err = authn.NewStore([]string{}...)
+	if err != nil {
+		return err
+	}
+
+	// Create temp dir where to put pulled artifacts
+	tmpDir, err := os.MkdirTemp("", "falcoctl")
+	if err != nil {
+		return fmt.Errorf("cannot create temporary directory: %w", err)
+	}
+	defer os.RemoveAll(tmpDir)
+
+	// Install artifacts
+	for _, name := range args {
+		var ref string
+		if strings.ContainsAny(name, ":@") {
+			ref = name
+		} else {
+			entry, ok := mergedIndexes.EntryByName(name)
+			if !ok {
+				o.Printer.Warning.Printf("cannot find %s among the configured indexes, skipping\n", name)
+				continue
+			}
+			ref = fmt.Sprintf("%s/%s:latest", entry.Registry, entry.Repository)
+		}
```

Comment:
I believe we need a simple parser for this.

Reply:
I have added a dedicated logic for this